subtests/docker_cli/build/build.py

```diff
@@ -114,213 +124,191 @@ def dockerfile_repo_replace(self, path, repo):
-            for cont in created_containers[:-1]:     # All but one exist
-                conts = [_.long_id for _ in containers if _.cmp_id(cont)]
-                self.failif(len(conts) != 0, "Intermediary container '%s' is "
-                            "present although it should been removed by build"
-                            "\n%s" % (cont, containers))
-            # Last one should not
-            conts = [_.long_id for _ in containers
-                     if _.cmp_id(created_containers[-1])]
-            self.failif(len(conts) != 1, "Intermediary container '%s' not "
-                        "present once in containers\n%s"
-                        % (created_containers[-1], containers))
-        elif build_def.get('intermediary_containers'):    # should be preserved
-            for cont in created_containers:
-                conts = [_.long_id for _ in containers if _.cmp_id(cont)]
-                self.failif(len(conts) != 1, "Intermediary container '%s' not "
-                            "present once in containers\n%s" % (cont,
-                                                                containers))
-        else:   # should not be present
-            for cont in created_containers:
-                conts = [_.long_id for _ in containers if _.cmp_id(cont)]
-                self.failif(len(conts) != 0, "Intermediary container '%s' is "
-                            "present although it should been removed by build"
-                            "\n%s" % (cont, containers))
-        self.logdebug("%s:\t%s intermediary containers\tOK",
-                      build_def['image_name'], len(created_containers))
-
-    def _postprocess_output(self, build_def):
-        """
-        Checks the presence of messages in the output
-        """
-        result = build_def['result']
-        exp = build_def.get('stdout')
-        self.failif(exp and not re.search(exp, result.stdout, re.M), "Expected"
-                    " message '%s' not found in build stdout:\n%s"
-                    % (exp, result))
-        exp = build_def.get('no_stdout')
-        self.failif(exp and re.search(exp, result.stdout, re.M), "Forbidden "
-                    "message '%s' was found in build stdout:\n%s"
-                    % (exp, result))
-        self.logdebug("%s:\tOutput messages\tOK", build_def['image_name'])
-
-    def _postprocess_no_containers(self):
-        """
-        Check that during test run expected number of containers were created
-        """
+        dkrimgs.images_args = DockerImages.images_args
+
+        img_names = [di.full_name for di in images]
+        imf_ids = [di.short_id for di in images]
+        self.failif(build_def.image_name not in img_names,
+                    "Test image '%s' not found after build"
+                    % build_def.image_name)
+
+        stdout = build_def.dockercmd.stdout
+        created_ids = [mobj.group(1)
+                       for mobj in RE_IMAGES.search(stdout)
```

Comment:
Ahh, okay, right the match-object returned by search cannot be looped over.  I think what we want here is ``finditer`` instead of ``search``, thanks I'll check it out...